includes/notes/class-wc-admin-notes-order-milestones.php

```diff
@@ -124,6 +124,11 @@ public function get_orders_count( $no_cache = false ) {
 	 * Used to avoid celebrating milestones that were reached before plugin activation.
 	 */
 	public function backfill_last_milestone() {
+		// If the milestone notes have been disabled via filter, bail.
+		if ( ! $this->are_milestones_enabled() ) {
```

Comment:
What do you think about putting this in the constructor instead so we avoid scheduling events and adding hooks entirely?

Reply:
Potentially too early? A lot of the constructors load as the plugins load, so unless your plugin/code runs before then the filters will be added.

It's probably ok to be here so that if it's added via other actions, or called manually, the new filter still applies.

Reply:
Indeed per @mikejolley's comment - I hit issues in testing with the filter in the constructor, and also since the method could be called manually it seemed like having the check here would be safest.